src/uct/ib/rc/base/rc_ep.h

```diff
@@ -353,7 +351,8 @@ uct_rc_txqp_completion_inl_resp(uct_rc_txqp_t *txqp, const void *resp, uint16_t
 static UCS_F_ALWAYS_INLINE uint8_t
 uct_rc_iface_tx_moderation(uct_rc_iface_t *iface, uct_rc_txqp_t *txqp, uint8_t flag)
 {
-    return (txqp->unsignaled >= iface->config.tx_moderation) ? flag : 0;
+    return ((txqp->unsignaled >= iface->config.tx_moderation) ||
```

Comment:
@evgeny-leksikov sounds good, we need to also put the check for cq credits under that `if' to avoid overhead:
```
if (unlikely(cq_available < MAX_RES_COUNT)) {
   if (cq_available == 0) { 
       return UCS_ERR_NO_RESOURCE;  // use UCT_RC_CHECK_CQE macro
   };
   txqp->unsignaled_real = txqp->unsignaled; 
   txqp->unsignaled = UINT16_MAX;
}
```